client/src/main/scala/com.gu.contentapi.client/model/PaginatedApiResponse.scala

```diff
@@ -0,0 +1,35 @@
+package com.gu.contentapi.client
+
+import com.gu.contentapi.client.model.v1._
+import com.gu.contentatom.thrift.Atom
+
+/** Typeclass witnessing how to extract the id for
+  * a paginated query
+  */
+trait PaginatedApiResponse[Response] {
+  /** the id for a next query, if any */
+  def getNextId: Response => Option[String]
+}
+
+/** Instances of [[PaginatedApiResponse]] for response types
+  * potentially containing pages of results
+  */
+object PaginatedApiResponse {
+
+  implicit val searchResponse = new PaginatedApiResponse[SearchResponse] {
+    def getNextId = r => if (r.results.length < r.pageSize) None else r.results.lastOption.map(_.id)
+  }
+
+  implicit val itemResponse = new PaginatedApiResponse[ItemResponse] {
+    def getNextId = r => if (r.results.exists(res => r.pageSize.exists(res.length < _))) None else r.results.flatMap(_.lastOption.map(_.id))
```

Comment:
What does the `pageSize` mean in an `ItemResponse`?

Reply:
If the item is a tag for instance, it will return [a list of results](https://github.com/guardian/content-api-models/blob/master/models/src/main/thrift/content/v1.thrift#L1681) for that tag.

Reply:
Are these tag results or content (article) results?

Reply:
Articles. Yeah I should add normalisation rules so that information is not lost. For example, if the item is a tag, the next query should have the `tag` parameter set.

Reply:
Yes. It would be surprising to make a query for tags, call getNext, and get articles in the results!

Reply:
I'll put this feature on hold, it needs some thinking.